CGE_equations.py

```diff
@@ -0,0 +1,688 @@
+        pWe (1D numpy array): The world price of commodity i in foreign currency
+
+    Returns:
+        pe (1D numpy array): Price of commodity i exports in domestic currency
+    '''
+    pe = er * pWe
+    return pe
+
+
+def eqpm(er, pWm):
+    '''
+    Import prices.
+
+    .. math::
+        pm_{i} = \varepsilon \cdot pWm_{i}
+
+    Args:
+        er (float): The real exchange rate
+        pWm (1D numpy array): The world price of commodity i in foreign currency.
+
+    Returns:
+        pm (1D numpy array): The price of commodity i imports in domestic currency.
+    '''
+    pm = er * pWm
+    return pm
+
+
+def eqbop(pWe, pWm, E, M, Sf, Fsh, er):
+    '''
+    Balance of payments.
+
+    .. math::
+        \sum_{i}pWe_{i}E_{i} + \frac{Sf}{\varepsilon} = \sum_{i}pWm_{i}M_{i} + \frac{Fsh}{\varepsilon}
+
+    Args:
+        pWe (1D numpy array): The world price of commodity i in foreign currency
+        pWm (1D numpy array): The world price of commodity i in foreign currency.
+        E (1D numpy array): Exports of commodity i
+        M (1D numpy array): Imports of commodity i
+        Sf (float): Total foreign savings (??)
+        Fsh = Repatriated profits
+        er (float): The real exchange rate
+
+    Returns:
+        bop_error (float): Error in balance of payments equation.
+
+    '''
+    bop_error = (pWe * E).sum() + Sf / er - ((pWm * M).sum() + Fsh / er)
+    return bop_error
+
+
+def eqSf(g, lam, pq, Kf):
+    '''
+    Net foreign investment/savings.
+
+    .. math::
+        Sf = g Kf \sum_{j} \lambda_{j} pq_{j}
+
+    Args:
+        g (float): Exogenous long run growth rate of the economy
+        lam (1D numpy array): Fixed shares of investment for each good j
+        pq (1D numpy array): price of XXXX for each good i
+        Kf (float): Foreign owned capital ??
+
+    Returns:
+        Sf (float): Total foreign savings (??)
+    '''
+    Sf = g * Kf * (lam * pq).sum()
+    return Sf
+
+
+def eqQ(gamma, deltam, deltad, eta, M, D):
+    '''
+    CES production function for the importing firm.
+
+    .. math::
+        Q_{i} = \gamma_{i}\left[\delta^{m}_{i}M^{\eta_{i}}_{i} + \delta^{d}_{i}D^{\eta_{i}}_{i}\right]^{\frac{1}{\eta_{i}}}
+
+    Args:
+        gamma (1D numpy array): ??
+        deltam (1D numpy array): Share parameter for use of imports of good i in produciton Armington good i
+        deltad (1D numpy array): Share parameter for use of domestically produced good i in produciton Armington good i
+        eta (1D numpy array): The elasticity of substitution between imports and domestically supplied good i
+        M (1D numpy array): The imports of good i
+        D (1D numpy array): The domestic supply of good i from domestic production
+
+    Returns:
+        Q (1D numpy array): The domestic supply of good i, the Armington good
+    '''
+    Q = gamma * (deltam * M ** eta + deltad * D ** eta) ** (1 / eta)
+    return Q
+
+
+def eqM(gamma, deltam, deltad, eta, Q, pq, pm, taum):
```

Comment:
@DPrihardini I didn't see this equation in the slides.  Is the grouping of terms correct?

Reply:
What do you mean by grouping of terms?

Reply:
I guess I'm wondering if in the equation on line 513 - is just that term in the denominator under `pq` raised to the `1/(1-eta)`?  I wasn't sure what equation this was derived from.

Reply:
The grouping of terms in line 513 is correct. It should be the entire expression in brackets starting from gamma that is raised to 1/(1-eta). It is the first order condition for imports if you solve the problem in slide 28.